js/CADWorker/CascadeStudioWorkerState.js

```diff
@@ -0,0 +1,26 @@
+export const messageHandlers = {};
+
+export let argCache = {};
+export let currentLineNumber = 0;
+export let currentOp = "";
+export let currentShape;
+export let externalShapes = {};
+export let GUIState = {};
+export let oc = null;
+export let opNumber = 0; // This keeps track of the progress of the evaluation
+export let usedHashes = {};
+
+export const setArgCache = val => (argCache = val);
+export const setCurrentOp = val => (currentOp = val);
+export const setCurrentShape = val => (currentShape = val);
+export const setCurrentLineNumber = val => (currentLineNumber = val);
+export const resetExternalShapes = () => (externalShapes = {});
+export const setGUIState = val => (GUIState = val);
+export const setOc = ocInit => (oc = ocInit);
+export const setOpNumber = val => (opNumber = val);
+export const setUsedHashes = val => (usedHashes = val);
```

Comment:
### A little background on modules and how vars are handled
(you probably already know but don't want to make assumptions).
Variables that are exported directly from modules can be used in other modules and are essentially getters for that variable in the original module. The variable can update, but only from within the module, and by update I mean re-assigned, of-course due to the way memory management works Arrays and Objects can be mutated outside of the original modules and will flow on to everywhere they are imported too.

### So what have I done here?
Firstly my primary motive was optimize for the smallest diff in existing files as possible to lower the chance or severity of future conflicts (since this is a big change and there are other PR's currently up), This is apposed to optimizing for best practices or style. So in the case of global variables, I wanted export variables of the same name so everywhere in the code where these variables are read or mutated nothing would need to change, it's only in places where the variable needs to be re-assigned that the "setters" I've added here need to be used and will cause a diff.

That explains why I've chosen this particular method. The exact same explanation applies to `js/MainPage/CascadeState.js`. `CascadeState` is for the original main thread globals, and this file is for worker Globals.

Reply:
I appreciate the effort here; thank you.